include/eve/algo/container/soa_vector.hpp

```diff
@@ -0,0 +1,277 @@
+
+    //! Checks if the container has no elements
+    EVE_FORCEINLINE bool empty() const noexcept { return size() == 0ULL; }
+
+    //! @brief Increase the capacity of the vector to a value that's greater or equal to `n`.
+    //! If `n` is greater than the current capacity(), new storage is allocated and all iterators,
+    //! including the past-the-end iterator, and all references to the elements are invalidated.
+    //! Otherwise, the method does nothing.
+    //! @warning reserve() does not change the size of the vector.
+    //! @param n New capacity of the `eve::soa_vector`
+    void reserve(std::size_t n)
+    {
+      kumi::for_each([&](auto& m) { m.reserve(n); }, storage);
+    }
+
+    //==============================================================================================
+    //! @}
+    //==============================================================================================
+
+    //==============================================================================================
+    //! @name Modifiers
+    //! @{
+    //==============================================================================================
+    //! @brief Clear the contents of the container
+    void clear() { kumi::for_each([&](auto& m) { m.clear(); }, storage); }
+
+    //! @brief Appends the given element value to the end of the container.
+    //! If the new size() is greater than capacity() then all iterators and references (including
+    //! the past-the-end iterator) are invalidated. Otherwise only the past-the-end iterator is
+    //! invalidated.
+    //! @param value [Value](@eve::vectorizable) to append
+    EVE_FORCEINLINE void push_back(value_type const& value) noexcept
+    {
+      kumi::for_each( [&](auto& m, auto const& v) { m.push_back(v); }
+                    , storage, kumi::flatten_all(value)
+                    );
+    }
+
+    //! @brief Removes the last element of the container.
+    //! Calling pop_back on an empty container results in undefined behavior.
+    //! Iterators and references to the last element, as well as the end() iterator, are invalidated.
+    EVE_FORCEINLINE void pop_back() noexcept
+    {
+      kumi::for_each([&](auto& m) { m.pop_back(); }, storage);
+    }
+
+    //! @brief Resizes the container to contain count elements.
+    //!
+    //! If the current size is greater than `n`, the container is reduced to its first `n` elements.
+    //! If the current size is less than `n`, additional elements are appended. Those elements are
+    //! either default-inserted or copied if `value` is provided.
+    //! @param n      new size of the container
+    //! @param value  the value to initialize the new elements with
+    void resize(std::size_t n, value_type value = {})
+    {
+      kumi::for_each( [&](auto& m, auto const& v) { m.resize(n,v); }
+                    , storage, kumi::flatten_all(value)
+                    );
+    }
```

Comment:
shrink_to_fit? Maybe capacity stuff when we have a custom storage implementation.

Reply:
I added shrink_to_fit too. But here, we just want to call resize. We'll be clever as you say with the compact mono-vector implementation later on